src/libraries/Common/src/Interop/Android/System.Security.Cryptography.Native.Android/Interop.X509.cs

```diff
@@ -10,23 +10,33 @@ internal static partial class Interop
 {
     internal static partial class AndroidCrypto
     {
+        private const int INSUFFICIENT_BUFFER = -1;
+        private const int SUCCESS = 1;
+
         [DllImport(Libraries.CryptoNative, EntryPoint = "AndroidCryptoNative_X509Decode")]
         internal static extern SafeX509Handle X509Decode(ref byte buf, int len);
 
         [DllImport(Libraries.CryptoNative, EntryPoint = "AndroidCryptoNative_X509Encode")]
-        private static extern int X509Encode(SafeX509Handle x, [Out] byte[]? buf, int len);
+        private static extern int X509Encode(SafeX509Handle x, [Out] byte[]? buf, ref int len);
         internal static byte[] X509Encode(SafeX509Handle x)
         {
-            return GetDynamicBuffer((ptr, buf, i) => X509Encode(ptr, buf, i), x);
+            int len = 0;
+            int ret = X509Encode(x, null, ref len);
+            if (ret != INSUFFICIENT_BUFFER)
+                throw new CryptographicException();
+
+            byte[] encoded = new byte[len];
+            ret = X509Encode(x, encoded, ref len);
+            if (ret != SUCCESS)
```

Comment:
I assume the answer is "no", but is there any chance len here might != encoded.Length, i.e. on the previous call we got one length number but here we got a different one?

Reply:
They should be the same (as long as `ret == SUCCESS`). The properties of the object being encoded can't be changed so the encoded length wouldn't change between the previous call and this one.